lib/luaxtable.c

```diff
@@ -101,13 +101,12 @@ static int luaxtable_pushparams(lua_State *L, const struct xt_action_param *par,
 	return 0;
 }
 
-#define luaxtable_call(L, op, xtable, skb, par, opt) \
-	luaxtable_pushparams(L, par, xtable, skb, opt) == -1 || \
-	luaxtable_docall(L, xtable, op, 2, 1) == -1
+#define luaxtable_call(L, op, xtable, skb, par, opt)	\
+	((luaxtable_pushparams(L, par, xtable, skb, opt) == -1) || (luaxtable_docall(L, xtable, op, 2, 1) == -1))
 
 static int luaxtable_domatch(lua_State *L, luaxtable_t *xtable, const struct sk_buff *skb, struct xt_action_param *par, int fallback)
 {
-	if (luaxtable_call(L, "match", xtable, skb, par, LUADATA_OPT_READONLY) != 0)
+	if (luaxtable_call(L, "match", xtable, (struct sk_buff *)skb, par, LUADATA_OPT_READONLY) != 0)
```

Comment:
We are getting a `const struct sk_buff *` at the match handler, is it fine to discard it and use `skb_linearize` on this ?

Reply:
It's resolved